state_transfer/meta/interfaces.go

```diff
@@ -0,0 +1,50 @@
+package meta
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// NamespacedNamePair knows how to map a source and a destination namespace involved in state transfer
+type NamespacedNamePair interface {
+	// Source represents source namespace and name
+	Source() types.NamespacedName
+	// Destination represents destination namespace and name
+	Destination() types.NamespacedName
+}
+
+// Mutation allows mutating state transfer resources on the fly based on the type
+type Mutation interface {
+	Type() MutationType
+}
+
+// PodSpecMutation knows how to mutate PodSpec fields of a corev1.Pod resource
+type PodSpecMutation interface {
```

Comment:
we talked about simplifying this implementation on slack, just wanted to make sure it is captured, can you please create an issue for follow-up or fix it here? either works for me

Suggestion:

1. convert the `Mutation` interface similar to the option interface with `AppliesTo` or `Mutate`method
2. create two func types, `type MutatePodSpec(*corev1.Pod) error` and `type MutateResourceMetadata (metav1.Metadata) error`. Implement Mutation interface from those functions by simple calling the instance of func type

I think it will simplify a lot of the implementation and we can create utility functions for those types like, SCC mutator, node selector mutator, node name mutator etc that consumers of lib can choose to use as they so wish.